### DISCUSSION TITLE
Confusing LinkWidget behavior

### DISCUSSION THREAD

**Discussion by CodaCodr** [TiddlyWiki/TiddlyWiki5#5391] [Q&A] [answered]
Before I post this as a bug...

I had an issue this morning which kind of surprised me - surprised in the sense that I haven't seen this behavior before. In the following code... 

1. The span title attribute using `<<link>>` is set correctly.
2. The rendered `href` of link 1 using the same `<<link>>` is wrong.
3. The rendered `href` of link 2 using the same `<<link>>` but different `textContent` is correct.

I rewrote enough of my local code to make a simpler example that works on tiddlywiki.com.

```
<$set name=link value={{{ [prefix[Hello]] }}}>
 <span title=<<link>>>
  1:<$link to=<<link>>>{{{ [<link>split[]first[5]join[]] }}}</$link>
 </span>
 2:<$link to=<<link>>>Hello</$link>
</$set>
```

Does the transclusion/filter affect the href construction? Crazy... but it seems to. :/ 

![image](https://user-images.githubusercontent.com/436812/104225662-4c37d400-540c-11eb-907c-21f0be07438f.png)

This works, but I don't see why it's necessary here - can someone explain, please?

```
  <$link to=<<link>>><$text text={{{ [<link>split[]first[5]join[]] }}}/></$link>
```

**Comment by saqimtiaz** [comment, marked as answer]
`{{{ [<link>split[]first[5]join[]] }}}` is the issue. 

Filtered transclusions always generate a link unless a different template is specified.

You can use `<$text text={{{ [<link>split[]first[5]join[]] }}} />` instead.

**Comment by saqimtiaz** [comment]
![image](https://user-images.githubusercontent.com/68092/104226805-a8b8e480-5448-11eb-87d9-17e43a8ad3ac.png)

Is what the DOM looks like for the code you posted. See the nested links?

**Reply by CodaCodr** [reply]
Bingo. Thanks, @saqimtiaz.

Yes, I used the `<$text...>` approach above (guess you leapt into a reply and didn't notice).
